src/main/kotlin/lgp/core/evolution/ResultsExporter.kt

```diff
@@ -0,0 +1,339 @@
+
+        override val results: MutableList<ExportableResult<T>> = mutableListOf()
+
+        /**
+         * Add a single result.
+         */
+        override fun add(result: ExportableResult<T>) {
+            this.results.add(result)
+        }
+
+        /**
+         * Add a collection of results.
+         */
+        override fun addAll(results: List<ExportableResult<T>>) {
+            this.results.addAll(results)
+        }
+
+        /**
+         * Clear all stored results.
+         */
+        override fun close() {
+            this.results.clear()
+        }
+
+        /**
+         * Returns a representation of any currently aggregated results in a simple table format.
+         */
+        override fun toString(): String {
+            if (this.results.isEmpty())
+                return "No results."
+
+            val table = StringBuilder()
+            val firstResult = this.results.first().export()
+            val header = firstResult.map { result -> result.first }.toList()
+
+            header.map { key -> table.append("%-28s".format(key)) }
+            table.appendln()
+
+            this.results.forEach { result ->
+                val exported = result.export()
+
+                exported.map { value -> table.append("%-28s".format(value.second)) }
+                table.append('\n')
+            }
+
+            return table.toString()
+        }
+
+        /**
+         * Provides information about the module.
+         */
+        override val information = ModuleInformation(
+                description = "Aggregates results that are able to be exported."
+        )
+    }
+
+    /**
+     * A [ResultAggregator] that buffers results and flushes to a CSV output regularly.
+     *
+     * This implementation will avoid using large amounts of memory as any results aggregated
+     * are regularly written out to file, instead of being kept in memory. Currently, the only
+     * supported [ResultOutputProvider] is [ResultOutputProviders.CsvResultOutputProvider] as
+     * no other output provider implementations support the buffered output required.
+     *
+     * Note that the buffering is done in a thread-safe way to ensure that there is no contention
+     * between threads when results are being added/output simultaneously. Despite this, there is no
+     * guarantee that the order of the results will be sequential in a multi-threaded environment. The
+     * built-in [Trainers] ensure that the results will be grouped by their run however.
+     *
+     * @param outputProvider A [ResultOutputProviders.CsvResultOutputProvider] instance.
+     * @param bufferSize How many results to keep in memory before writing to [outputProvider].
+     * @param verbose Writes detailed output about any buffering operations to standard out.
+     */
+    class BufferedResultAggregator<T>(
+            private val outputProvider: ResultOutputProviders.CsvResultOutputProvider<T>,
+            private val bufferSize: Int = 100,
```

Comment:
Could potentially increase the default buffer size? In practice I'd expect that users would set it to some fraction of the total number of generations, so that for example results are flushed `n` times per run -- so maybe a low default is fine.

Reply:
Going to keep it low as it user-definable.